lib/ansible/module_utils/k8s_common.py

```diff
@@ -0,0 +1,290 @@
+#
+#  Copyright 2017 Red Hat | Ansible
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+import copy
+import json
+import os
+import yaml
+
+from ansible.module_utils.basic import AnsibleModule
+from openshift.helper.ansible import AnsibleModuleHelper, OpenShiftException, ARG_ATTRIBUTES_BLACKLIST
+
+
+class OpenShiftAnsibleException(OpenShiftException):
+    pass
+
+
+class OpenShiftAnsibleModule(AnsibleModule):
+
+    def __init__(self, kind, api_version):
+        self.api_version = api_version
+        self.kind = kind
+        self.argspec_cache = None
+
+        try:
+            self.helper = AnsibleModuleHelper(api_version, kind)
+        except OpenShiftException as exc:
+            raise OpenShiftAnsibleException(message=exc.message)
+
+        mutually_exclusive = (
+            ('resource_definition', 'src'),
+        )
+
+        AnsibleModule.__init__(self,
+                               argument_spec=self.argspec,
+                               supports_check_mode=True,
+                               mutually_exclusive=mutually_exclusive)
+
+    @property
+    def argspec(self):
+        """
+        Build the module argument spec from the helper.argspec, removing any extra attributes not needed by
+        Ansible.
+
+        :return: dict: a valid Ansible argument spec
+        """
+        if self.argspec_cache:
+            return self.argspec_cache
```

Comment:
`if not self.argspec_cache:` allows you to have a single return, not blocker, just seems cleaner